test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs

```diff
@@ -4095,6 +4095,99 @@ public async Task Restore_WithHttpSourceAndAllowInsecureConnectionsFalse_ThrowsE
             Assert.Contains(httpSourceUrl, logMessage.Message);
         }
 
+        [Theory]
+        [InlineData("8.0.100")]
```

Comment:
When i think of the different codepaths this theory test exercises, it's really only testing the parsing method and the compare. 
That's well tested through unit tests.
I'm not really sure we need to test all these different versions. 1 version is enough.